bundles/org.openhab.binding.enphase/src/main/java/org/openhab/binding/enphase/internal/handler/EnvoyBridgeHandler.java

```diff
@@ -0,0 +1,410 @@
+                    updateState(channelUID, new QuantityType<>(data.wattHoursToday, Units.WATT_HOUR));
+                    break;
+                case ENVOY_WATT_HOURS_SEVEN_DAYS:
+                    updateState(channelUID, new QuantityType<>(data.wattHoursSevenDays, Units.WATT_HOUR));
+                    break;
+                case ENVOY_WATT_HOURS_LIFETIME:
+                    updateState(channelUID, new QuantityType<>(data.wattHoursLifetime, Units.WATT_HOUR));
+                    break;
+                case ENVOY_WATTS_NOW:
+                    updateState(channelUID, new QuantityType<>(data.wattsNow, Units.WATT));
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(EnphaseDevicesDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(EnvoyConfiguration.class);
+        if (!EnphaseBindingConstants.isValidSerial(configuration.serialNumber)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Serial Number is not valid");
+            return;
+        }
+        if (configuration.hostname.isEmpty()) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Waiting to retrieve ip address of the envoy gateway. Can take up to a minute.");
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        connector.setConfiguration(configuration);
+        consumptionSupported = FeatureStatus.UNKNOWN;
+        jsonSupported = FeatureStatus.UNKNOWN;
+        invertersCache = new ExpiringCache<>(Duration.of(configuration.refresh, ChronoUnit.MINUTES),
+                this::refreshInverters);
+        devicesCache = new ExpiringCache<>(Duration.of(configuration.refresh, ChronoUnit.MINUTES),
+                this::refreshDevices);
+        updataDataFuture = scheduler.scheduleWithFixedDelay(this::updateData, 0, configuration.refresh,
+                TimeUnit.MINUTES);
+    }
+
+    /**
+     * Method called by the ExpiringCache when no inverter data is present to get the data from the Envoy gateway.
+     * When there are connection problems it will start a scheduled job to try to reconnect to the
+     *
+     * @return the inverter data from the Envoy gateway or null if no data is available.
+     */
+    private @Nullable Map<String, @Nullable InverterDTO> refreshInverters() {
+        try {
+            return connector.getInverters().stream()
+                    .collect(Collectors.toMap(InverterDTO::getSerialNumber, Function.identity()));
+        } catch (final EnvoyNoHostnameException e) {
+            // ignore hostname exception here. It's already handled by others.
+        } catch (final EnvoyConnectionException e) {
+            logger.trace("refreshInverters connection problem", e);
```

Comment:
It seems unreasonable to log the stacktrace when the network fails.

Reply:
It's therefor logged to trace. To have the ability to get debug information just in case something explainable is triggered.